src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/util/FeatureToggleConditionExtension.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.professionalapi.util;
+
+import com.launchdarkly.sdk.server.LDClient;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.hmcts.reform.professionalapi.service.impl.FeatureToggleServiceImpl;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.Optional;
+
+import static java.lang.System.getenv;
+import static org.apache.commons.lang3.BooleanUtils.isNotTrue;
+import static org.junit.platform.commons.util.AnnotationUtils.findAnnotation;
+
+public class FeatureToggleConditionExtension implements ExecutionCondition {
+
+    private static FeatureToggleServiceImpl featureToggleService;
+
+    private static boolean isInitialized = false;
+
+    private static String flagName;
+
+    @Override
+    @SuppressWarnings("checkstyle:Indentation")
+    public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext context) {
+
+        if (isNotTrue(isInitialized)) {
+            initialize();
+        }
+
+        final Optional<AnnotatedElement> element = context.getElement();
+
+        return element.map(annotatedElement -> {
+                    Optional<ToggleEnable> toggleEnable = findAnnotation(element, ToggleEnable.class);
+
+                    return toggleEnable.map(toggle -> {
+                        featureToggleService.mapServiceToFlag();
+                        flagName = featureToggleService.getLaunchDarklyMap().get(toggle.mapKey());
+
+                        final boolean isFlagEnabled =
+                                featureToggleService
+                                        .isFlagEnabled("rd_user_profile_api", flagName);
+
```

Comment:
can you check the service name ?  .isFlagEnabled("rd_user_profile_api", flagName);

Reply:
Good catch - Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/2402aca945bf5e65dcbaaa0f0784c154e46c21fd 👍🏼